### ISSUE TITLE
400: Please fill out the access category and reason correctly

### ISSUE THREAD

**Issue by RankoR** [getsentry/self-hosted#1442]
### Version

22.4.0

### Steps to Reproduce

1. Go to admin (https://host/manage/)
2. Enter the required password

### Expected Result

Logged in successfully

### Actual Result

Request:

```
curl 'https://host/api/0/auth/' \
  -X 'PUT' \
  -H 'authority: host' \
  -H 'accept: application/json; charset=utf-8' \
  -H 'accept-language: en-US,en;q=0.9,ru;q=0.8' \
  -H 'content-type: application/json' \
  -H 'cookie: sudo=replaced; sc=replaced; sentrysid=replaced' \
  -H 'dnt: 1' \
  -H 'origin: https://host' \
  -H 'referer: https://host/manage/' \
  -H 'sec-ch-ua: "Chromium";v="100", " Not A;Brand";v="99"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.147 Safari/537.36' \
  -H 'x-csrftoken: replaced' \
  --data-raw '{"isSuperuserModal":true,"password":"replaced"}'
```

Response:

```
< HTTP/2 400 
< server: nginx
< date: Fri, 06 May 2022 18:42:12 GMT
< content-type: application/json
< content-length: 101
< access-control-allow-origin: https://host
< access-control-allow-methods: GET, POST, PUT, DELETE, HEAD, OPTIONS
< allow: GET, POST, PUT, DELETE, HEAD, OPTIONS
< access-control-allow-headers: X-Sentry-Auth, X-Requested-With, Origin, Accept, Content-Type, Authentication, Authorization, Content-Encoding
< access-control-expose-headers: X-Sentry-Error, Retry-After
< x-sentry-rate-limit-remaining: 369
< x-sentry-rate-limit-limit: 370
< x-sentry-rate-limit-reset: 1651862533
< x-sentry-rate-limit-concurrentremaining: 299
< x-sentry-rate-limit-concurrentlimit: 300
< vary: Accept-Language, Cookie
< content-language: en
< x-frame-options: deny
< x-content-type-options: nosniff
< x-xss-protection: 1; mode=block
< set-cookie: sudo=replaced; expires=Fri, 06 May 2022 21:42:12 GMT; HttpOnly; Max-Age=10800; Path=/
< set-cookie: sc=replaced; expires=Fri, 05 May 2023 18:42:12 GMT; Max-Age=31449600; Path=/; SameSite=Lax
< set-cookie: sentrysid=replaced; expires=Fri, 20 May 2022 18:42:12 GMT; HttpOnly; Max-Age=1209600; Path=/
< x-frame-options: SAMEORIGIN
< x-content-type-options: nosniff
< x-xss-protection: 1; mode=block
< 
{"superuserAccessCategory":["This field is required."],"superuserReason":["This field is required."]}%   
```

**Comment by chadwhitacre** [comment]
Hrm. @getsentry/enterprise What's the expected behavior of the super-user modal on self-hosted? Did we consider that case? 🙈

**Comment by maxiuyuan** [comment]
> Hrm. @getsentry/enterprise What's the expected behavior of the super-user modal on self-hosted? Did we consider that case? 🙈
This shouldn't be enabled for self-hosted. I'll be looking into this

**Comment by maxiuyuan** [comment]
Hi! I'm not able to replicate this. The error here happens on the superuser class if the superuser form isn't filled (not required for self-hosted) but we check if they're self-hosted before that check using `is_self_hosted()`. On my end, is_self_hosted() returns True and we skip the form validation and give superuser. Does this happen consistently? 

Also @chadwhitacre do u know if somehow our setting is faulty in this case and is_self_hosted() returned false

**Comment by RankoR** [comment]
I have a default docker installation (with the only difference in moving volumes to a non-default directory).
It happens consistently when I click here:
<img width="243" alt="image" src="https://user-images.githubusercontent.com/71142/167230849-09c6eab4-0ea2-445f-a9c0-add1c5e9944f.png">

And then click «continue» and «confirm» here:
<img width="604" alt="image" src="https://user-images.githubusercontent.com/71142/167230864-26b0c2a1-095c-4b73-b988-fcdd9584fa80.png">

**Comment by GTB3NW** [comment]
I have the problem with a fresh install. Can confirm no modifications my side beyond an override for docker-compose for something unrelated.

```
[root@sentry sentry]# grep -R SENTRY_SELF_HOSTED
[root@sentry sentry]# 
```

**Comment by GTB3NW** [comment]
I forgot to mention, I created a brand new account as a super user with no password set and was able to access it no problem, there could be another condition causing this

**Comment by chadwhitacre** [comment]
Hmm ... the [workaround from this ticket](https://github.com/getsentry/self-hosted/issues/1288) help with non-brand new accounts here as well? 🤔

```
docker exec -ti sentry-self-hosted_web_1 sentry permissions add -u <USER_EMAIL> -p "users.admin"
```

**Comment by maxiuyuan** [comment]
> Thanks @GTB3NW.
> 
> @maxiuyuan Are you testing with `sentry` dev env or with `self-hosted`?
> 
> (P.S. I am a bit slowed down by #914 or I would test.)

I tested with `sentry` dev env, I wasn't able to get `self-hosted` running

**Comment by Ovyerus** [comment]
Having this same issue with a new `self-hosted` deploy. The form doesn't seem to appear for some time after the initial setup, and I can access the admin pages just fine, but after a while (possibly after a restart?, given I've done a few due to config changes) the form pops up and I'm unable to access it. Trying to do another restart doesn't fix it.

**Comment by GigaFyde** [comment]
Just ran into this issue upgrading from 22.3.0 to 22.4.0
Like @GTB3NW said, logging out, and logging back in resolved the issue :)

**Comment by chadwhitacre** [comment]
> I wasn't able to get `self-hosted` running

@maxiuyuan Are you on M1, too?

**Comment by maxiuyuan** [comment]
> > I wasn't able to get `self-hosted` running
> 
> @maxiuyuan Are you on M1, too?

No, so it should work. Do you have time to help me set up self-hosted?

Also seeing that the ` admin page randomly started working again for a bit without the prompt` it makes me feel like our settings are being wonky

**Comment by chadwhitacre** [comment]
Yes, I'll follow up separately. tl;dr is [git clone self-hosted](https://github.com/getsentry/self-hosted) and [install.sh](https://develop.sentry.dev/self-hosted/) :]

**Comment by maxiuyuan** [comment]
I'm running `self-hosted` and can not replicate it still. To make sure my setup is the same here are the steps I took. I pulled https://github.com/getsentry/self-hosted version `22.4.0 `. I ran `./install.sh` and `docker compose up -d`. Then I went to http://localhost:9000/ and did the following.

https://user-images.githubusercontent.com/22259802/167949326-8cdd3434-71b8-484d-bb55-8833ad5ac9ce.mov

**Comment by chadwhitacre** [comment]
From the Discussions version of this thread:

@stoutZero:

> Hi,
> 
> In my case, I'm running self-hosted, behind nginx, I forwarded every request that came to the sentry subdomain via nginx,  
> to 127.0.0.1:9000 which should be `sentry-self-hosted-nginx-1` right?
> 
> Should I forward it to the Django app (`sentry-self-hosted-web-1`)?
> Or is that not possible?
> 
> Don't actually understand the difference between `0.0.0.0:9000->80/tcp` & `9000/tcp` in `docker ps` output.
> 
> I'm using `22.4.0` btw, just finished setting up 4 hours ago.

@stoutZero:

> After a bit of digging, seems like there are two redis instance in one vps, the first one is pre-existing on port `*:6379`,  
the second one is from `docker-compose.yml`, occupying this kind of forwarding: `0.0.0.0:6379->6379/tcp`.
> 
> That seems like a port conflict.
> 
> Is there a way so that the two can somehow co-exist?  
> Not particularly well versed in <s>hoodoo</s> docker magic...

@aminvakil:

> AFAICS redis in self-hosted does not expose any ports.
> 
> https://github.com/getsentry/self-hosted/blob/282c2692746996f89c79100c7fa8572e9f410db8/docker-compose.yml#L107-L118

@stoutZero:

> Well, okay, the error log of not being able to connect to redis, might be because redis service is not up yet.  
And yes, redis was not exposed, my bad.  
>
> I am still getting this error though:
>
> `[WARNING] django.request: Bad Request: /api/0/auth/ (status_code=400 request=<WSGIRequest: PUT '/api/0/auth/'>)`

@aminvakil:

> @stoutZero You should use 127.0.0.1:9000. The nginx container proxies requests to different containers itself.

@stoutZero:

> Yeah, that's what I did.
